.github/workflows/functional.yaml

```
@@ -21,8 +21,12 @@ jobs:
 
       - name: Set up Helm
         uses: azure/setup-helm@v4
-        with:
-          version: v3.11.3
+
+      - name: Set up chart testing
+        uses: helm/chart-testing-action@v2
+
+      - name: Run chart linting
+        run: ct lint --config ct.yaml
```

In the past I've noticed that the chart linter doesn't catch any templating errors or typos so it's worth adding a helm template check here too. 

It might also be worth piping that template output into [kubeconform](https://github.com/yannh/kubeconform) so that it validates some of the k8s resource schemas as well as the templating syntax. I haven't used kubeconform myself anywhere yet but it looks easy to install in CI so seems worth a try if it's easy to set up: https://github.com/yannh/kubeconform?tab=readme-ov-file#integrating-kubeconform-in-the-ci

```suggestion
      - name: Run chart linting
        run: ct lint --config ct.yaml
        
      - name: Run chart templating with default values
        run: helm template ci-test charts/operator
```